Mathlib/NumberTheory/Cyclotomic/Rat.lean

```diff
@@ -344,6 +344,91 @@ theorem subOneIntegralPowerBasis'_gen_prime [IsCyclotomicExtension {p} ℚ K]
     (hζ : IsPrimitiveRoot ζ ↑p) :
     Prime hζ.subOneIntegralPowerBasis'.gen := by simpa using hζ.zeta_sub_one_prime'
 
+open nonZeroDivisors in
+/-- The norm, relative to `ℤ`, of `ζ ^ p ^ s - 1` in a `p ^ (k + 1)`-th cyclotomic extension of `ℚ`
+is p ^ p ^ s` if `s ≤ k` and `p ^ (k - s + 1) ≠ 2`. -/
+lemma norm_toInteger_pow_sub_one_of_prime_pow_ne_two [IsCyclotomicExtension {p ^ (k + 1)} ℚ K]
```

Comment:
Using
```lean
set_option profiler true
set_option profiler.threshold 50
```
I see
```
typeclass inference of Algebra took 84.9ms
typeclass inference of Finite took 74.7ms
typeclass inference of NonUnitalSemiring took 52.3ms
typeclass inference of AddMonoidHomClass took 271ms
typeclass inference of MonoidHomClass took 151ms
typeclass inference of Algebra took 82.1ms
typeclass inference of Algebra took 59.8ms
typeclass inference of Module.Finite took 298ms
tactic execution of Lean.Parser.Tactic.rewriteSeq took 87.1ms
type checking took 71.2ms
```
See also below for some of the other declarations.
Maybe adding a few carefully chosen private local instances can speed up the file by quite a bit?
See [here](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/defLemma.20and.20docBlame.20linters.20on.20local.20instances.20in.20sections/near/433829610) for how to avoid triggering the linters.